crates/wasmtime/src/runtime/component/resource_table.rs

```diff
@@ -32,10 +33,10 @@ impl fmt::Display for ResourceTableError {
 impl core::error::Error for ResourceTableError {}
 
 /// The `ResourceTable` type maps a `Resource<T>` to its `T`.
-#[derive(Debug)]
 pub struct ResourceTable {
     entries: Vec<Entry>,
     free_head: Option<usize>,
+    debug: bool,
```

Comment:
Have you used per-table debugging in the past? If not how about moving this to a `const` in this module? That way it can still be easily flipped but won't take up space at runtime